src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -13,89 +14,57 @@ using namespace dnnl::impl::cpu::x64;
 namespace ov {
 namespace intel_cpu {
 
-inline static void transform_idxs_to_regs(const std::vector<size_t>& idxs, std::vector<Xbyak::Reg64>& regs) {
-    regs.resize(idxs.size());
-    std::transform(idxs.begin(), idxs.end(), regs.begin(), [](size_t idx){ return Xbyak::Reg64(static_cast<int>(idx)); });
-}
-
 /* ================== jit_loop_begin_emitter ====================== */
 
 jit_loop_begin_emitter::jit_loop_begin_emitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
                                                const ov::snippets::lowered::ExpressionPtr& expr)
-    : jit_emitter(h, isa), loop_begin_label{new Xbyak::Label()} {
-    in_out_type_ = emitter_in_out_map::gpr_to_gpr;
-}
-
-std::shared_ptr<ov::snippets::op::LoopEnd> jit_loop_begin_emitter::get_loop_end(const ov::snippets::lowered::ExpressionPtr& expr) {
-    OV_CPU_JIT_EMITTER_ASSERT(expr->get_output_port_connectors().size() == 1, "has invalid LoopBegin expression configuration");
-    const auto& consumers = expr->get_output_port_connector(0)->get_consumers();
-    OV_CPU_JIT_EMITTER_ASSERT(consumers.size() == 1, "has invalid LoopBegin expression configuration");
-    const auto loop_end = ov::as_type_ptr<snippets::op::LoopEnd>(consumers.cbegin()->get_expr()->get_node());
-    OV_CPU_JIT_EMITTER_ASSERT(loop_end != nullptr, "has invalid LoopBegin expression configuration");
-    return loop_end;
-}
-
-jit_loop_begin_static_emitter::jit_loop_begin_static_emitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
-                                                             const ov::snippets::lowered::ExpressionPtr& expr)
-    : jit_loop_begin_emitter(h, isa, expr) {
-    OV_CPU_JIT_EMITTER_ASSERT(ov::is_type<snippets::op::LoopBeginStatic>(expr->get_node()),
-                              "expects LoopBeginStatic expression");
-    const auto loop_end = ov::as_type_ptr<snippets::op::LoopEndStatic>(get_loop_end(expr));
+    : jit_emitter(h, isa), loop_begin_label{new Xbyak::Label()}, loop_end_label(nullptr) {
+    const auto loop_begin = ov::as_type_ptr<snippets::op::LoopBegin>(expr->get_node());
+    OV_CPU_JIT_EMITTER_ASSERT(loop_begin, "expects LoopBegin expression");
+    const auto loop_end = loop_begin->get_loop_end();
     work_amount = loop_end->get_work_amount();
     wa_increment = loop_end->get_increment();
     evaluate_once = loop_end->get_evaluate_once();
+    loop_id = loop_end->get_id();
+    is_work_amount_dynamic = ov::snippets::utils::is_dynamic_value(work_amount);
```

Comment:
Yes, it could be.
![image](https://github.com/openvinotoolkit/openvino/assets/43129309/2b5b3e99-b240-4180-a3aa-6beb8c0d2e22)
This case is handled on `loop_end_emitter` side. `loop_begin_emitter` is not interested in pointer arithmetic, its responsibility is to set correct `work_amount` to GPR: if it's unknown, put from runtime args, if it's defined, compile the value.

Reply:
M-hm, I see, thanks